### ISSUE TITLE
Switching Branches leaves files from previous branch

### ISSUE THREAD

**Issue by austinamorusocfc** [kactus-io/kactus#111]
## Description
Switching between branches leaves file from other branch in file system and in Kactus. 

## Version
* Kactus: Version 0.3.12
* Operating system: OSX Version 10.13.6

## Steps to Reproduce
![kactus](https://user-images.githubusercontent.com/39928550/44226959-5293ca00-a146-11e8-811d-f4f26c748d0c.gif)
![kactus2](https://user-images.githubusercontent.com/39928550/44226963-558eba80-a146-11e8-9cb8-86b7eb642e1a.gif)

### Expected Behavior

Only files from the branch are visible in Kactus and the file system. 

### Actual Behavior
Files from pervious branch are present after switching in current branch

## Additional Information
Makes Kactus unusable as designers don't understand whats happening. 

### Logs

[2018-07-25.kactus.production.log](https://github.com/kactus-io/kactus/files/2294766/2018-07-25.kactus.production.log)
[2018-07-30.kactus.production.log](https://github.com/kactus-io/kactus/files/2294767/2018-07-30.kactus.production.log)
[2018-08-09.kactus.production.log](https://github.com/kactus-io/kactus/files/2294768/2018-08-09.kactus.production.log)
[2018-08-16.kactus.production.log](https://github.com/kactus-io/kactus/files/2294769/2018-08-16.kactus.production.log)

**Comment by austinamorusocfc** [comment]
Work around setup git hook for post-commit to rm *.sketch

**Comment by austinamorusocfc** [comment]
git post-commit hook that runs this script seems to fix the issue

``` 
#!/bin/sh
[ "$SKIP_POST_CHECKOUT" = 1 ] && exit 0
rm -f -r ./*
SKIP_POST_CHECKOUT=1 git checkout .
```

**Comment by mathieudutour** [comment]
oh interesting, it's because sketch files are git ignored so they will stay on the disk when switching branches.

I'll think about a solution (I don't think the post-commit hook is correct since it's going to remove every files which is not what you want in the general case)

**Comment by austinamorusocfc** [comment]
It actually looks like all files well at the very least the folders that kactus make stick around from the pervious branch.

**Comment by riverbrandon** [comment]
Just logged #125 that I'm guessing is related to this.

**Comment by mathieudutour** [comment]
I think it's should be fixed in 0.3.17. Feel free to re-open if it's not
